aten/src/ATen/native/native_functions.yaml

```diff
@@ -3385,16 +3385,16 @@
     CPU, CUDA: logit_out
 
 - func: sin(Tensor self) -> Tensor
+  structured_delegate: sin.out
   variants: function, method
-  dispatch:
-    DefaultBackend: sin
```

Comment:
@ailzhang The functional and inplace variants of `sin` used to be marked `DefaultBackend`, but now we're codegen'ing CPU/CUDA specific versions of them. That sort of technically means that if a backend had implemented `sin_out` but not `sin` or `sin_`, it would break, right?

Do you think we need to label it BC-breaking? Or is this small enough of a risk not to bother (cc @bhosmer)

Reply:
It might also be worth discussing if this needs to be done on a case-by-case basis as I'm hoping to port a lot more of these and then it becomes a bigger issue.

Reply:
@bdhirsh We can make structured kernels automatically generate DefaultBackend kernels for the functional and inplace versions, unconditionally. This is easy to do and has no downsides.

Reply:
I'm going to through up a prototype of this

Reply:
(but I don't think we should block this PR on this, unless XLA is broken, which it doesn't look like it its)